File: quickselect/k-closest-points-to-origin.py
class Solution:
    def distance(self, x , y):
        return  math.sqrt(x ** 2 + y ** 2)

    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        ret = []
        heap = [] 
        for point in points:
            dist = self.distance(point[0], point[1]) 
            tup = (dist, point[0], point[1])
            heapq.heappush(heap, tup)
        for _ in range(k):
            min_ele = heapq.heappop(heap)
            ret.append([min_ele[1],min_ele[2]])
        return ret

File: geometry/k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        // Max-heap of size k 
        int[][] ret = new int[k][]; 
        PriorityQueue<int[]> pq = new PriorityQueue<>((a,b) -> {
            return Integer.compare(getDistance(b), getDistance(a)); 
        }); 
        for (int[] point: points){
            pq.add(point); 
            if (pq.size() > k){
                pq.poll(); 
            }
        }

        for(int i = 0; i < k; i++){
            int[] topK = pq.poll(); 
            ret[i] = topK; 
        }
        return ret; 

    }

    private int getDistance(int[] a){
        int distance = a[0] * a[0] + a[1] * a[1]; 
        return distance; 
    }
}